### DISCUSSION TITLE
Can I reuse an ObjectType to build an argument type take accepts OrderBy?

### DISCUSSION THREAD

**Discussion by brunolm** [MichalLytek/type-graphql#1350] [General]
I know I can reuse an `ObjecType` as an `InputType`. But then my class might have strings like in `name`.

What I want to achieve is the option to specify an "orderBy" that takes the exact same parameters as the `ObjectInput`, but the values should be `1` and `-1` or something like that.

Is it possible for me to reuse the exact same structure on my `ObjectType`, but have all props be numbers or something like that?

**Comment by MichalLytek** [comment]
No, there's no mapping utils like for raw TS types. However, you can use TS mapping types to describe the shape that the new order input type should implement, so you will have errors about missing fields when you change something in object type.
